README.md

```diff
@@ -1,57 +1,179 @@
-(Add steps to get up and running quickly)
+#### Setup
 
-1. git clone [repository clone url]
-2. cd [repository name]
-3. ...
+* Open a Bash terminal
+* `git clone` this repo
+* cd [repository name]
+* Open the project in Visual Studio Code `code .` and open a Bash terminal.
+* Log into your Azure account `az login [--tenant xxxx-xxx]` and set the default subscription `az account set -s <subscription>`
+* Prepare required environment variables to run the deployment scripts. We create a file in the folder `./temp` which is excluded from Git.
 
+  * Run the below to create the file `./temp/envvars.sh`, for `LOCATION` you can choose any Azure region supporting IoT Hub and Application Gateway.
 
-## Demo
+    ```bash
+    if [ ! -d "./temp" ]; then
+        mkdir ./temp
+    fi
+    >./temp/envvars.sh cat <<EOF
+    # change the below set to match your environment based on Readme
+    export TENANT_ID="xxx-xxxx-xxx-xxxx-xxxx"
+    export CERT_PASSWORD="xxx"
+    export LOCATION="westeurope"
+    export PREFIX="xxx"
+    export RESOURCE_GROUP="rg-xxx-xxx"
+    EOF
 
-A demo app is included to show how to use the project.
+    code ./temp/envvars.sh
+    ```
 
-To run the demo, follow these steps:
+  * The newly created `.sh` file should now open in Visual Studio Code.
+  * Edit the values to your preference, ensure `TENANT_ID` corresponds to your Azure tenant.
+  * For `PREFIX` use a short 5 character string that can be unique for the resource name composition.
+  * Load the variables by running:
 
-(Add steps to start up the demo)
+  ```bash
+  source ./temp/envvars.sh
+  ```
 
-1.
-2.
-3.
+#### Azure Resources with Virtual Network, private Storage and Application Gateway
 
-## Resources
+Deploy the Azure components for setting up Virtual Network, self-signed SSL certificate, Storage, Private Link, custom private DNS and Application Gateway configured to talk to the Storage account. This allows you to validate the flow before configuring Azure IoT Hub and client communication.
+
+* From the root directory of this repo, run the first part of the deployment. The script will use the environment variables and build composed resource names by appending the `PREFIX` variable as Azure resource names.
+
+```bash
+./deploy/quickstart-agw-storage.sh
+```
+
+* It will take a few minutes to deploy all resources. Keep the terminal open and take note of some of the generated DNS entries.
+* This scripts also uploads a `sample.txt` file to a Blob storage container and generates a SAS URI token for testing.
+* From a bash terminal, try out a `CURL` command to the DNS of the Public IP address attached to the Application Gateway.
+
+```bash
+curl --insecure "<APP_GATEWAY_SASURI copied output>"
+```
+
+* Run the same command with the Blob Public URI printed out by the script `BLOB_SASURI` value. Verify this does not succeed since this Storage account is blocking direct Internet traffic.
+
+* Configure your custom DNS and SSL certificate for end to end SSL encryption.
```

Comment:
The instructions below use a publicly accessible Key Vault to store the Pfx. You need to call out the security concerns with an internet-facing Key Vault that doesn't use private endpoints as a production warning.   (SFI teaching opportunity)